apps/api-extractor/src/collector/Collector.ts

```diff
@@ -147,6 +158,57 @@ export class Collector {
   }
 
   /**
+   * Searches the provided package.json for dependencies that match the provided package names and/or glob patterns
```

Comment:
@octogonz as it was originally written, `bundledPackages` matches could theoretically result in bundling of `devDependencies`. This doesn't seem semantically correct, and I didn't keep the behavior to start. This is another area where the spec for `bundledPackages` seems insufficient. Should we retain dev dep support here for backwards compatibility?

Reply:
What if a library webpacks its dependencies and rolls up the .d.ts files?  In that case the bundled library should be a `devDependency` so that callers don't need to install it.

Reply:
Based on our discussion, this is a case we expect existing consumers to be using. Will update to include dev, optional, and peer deps in the list we resolved wildcards against.